Remotion/Web/UnitTests/Core/ContentSecurityPolicy/CspEnabledHtmlTextWriterTest.cs

```diff
@@ -34,37 +35,47 @@ public class CspEnabledHtmlTextWriterTest
   {
     private HtmlHelper _htmlHelper;
     private CspEnabledHtmlTextWriter _writer;
-    private Mock<ISmartPageClientScriptManager> _clientScriptStub;
-    private Mock<ISmartPage> _pageStub;
+    private Mock<ICspClientScriptManager> _clientScriptStub;
     private Mock<INonceGenerator> _randomNumberGeneratorStub;
     private Mock<IRenderingFeatures> _renderingFeaturesStub;
     private Mock<IFallbackNavigationUrlProvider> _fallbackNavigationUrlProviderStub;
 
     [SetUp]
     public void SetUp ()
     {
-       _pageStub = new Mock<ISmartPage>(MockBehavior.Strict);
-      _clientScriptStub = new Mock<ISmartPageClientScriptManager>();
+      _clientScriptStub = new Mock<ICspClientScriptManager>();
       _randomNumberGeneratorStub = new Mock<INonceGenerator>(MockBehavior.Strict);
       _renderingFeaturesStub = new Mock<IRenderingFeatures>(MockBehavior.Strict);
       _renderingFeaturesStub.Setup(_ => _.EnableDiagnosticMetadata).Returns(false);
 
       _fallbackNavigationUrlProviderStub = new Mock<IFallbackNavigationUrlProvider>(MockBehavior.Strict);
 
-      _pageStub
-          .Setup(s => s.ClientScript)
-          .Returns(_clientScriptStub.Object);
-
       _htmlHelper = new HtmlHelper();
       _writer = new CspEnabledHtmlTextWriter(
-              _pageStub.Object,
+              _clientScriptStub.Object,
               _htmlHelper.Writer,
               _randomNumberGeneratorStub.Object,
               "TEST-NONCE",
               _renderingFeaturesStub.Object,
               _fallbackNavigationUrlProviderStub.Object);
     }
 
+    [Test]
+    public void Create_WithSmartPage_UsesAdapter ()
+    {
+      var writer = new CspEnabledHtmlTextWriter(
+          Mock.Of<ISmartPage>(),
+          _htmlHelper.Writer,
+          _randomNumberGeneratorStub.Object,
+          "TEST-NONCE",
+          _renderingFeaturesStub.Object,
+          _fallbackNavigationUrlProviderStub.Object);
+
+      Assert.That(
+          PrivateInvoke.GetNonPublicField(writer, "_scriptManager"),
+          Is.TypeOf<SmartPageCspClientScriptManagerAdapter>());
+    }
```

Comment:
The reflection call makes me sad since it is easy to just make the state public on the Writer. It's not enough to change this now, it's just... I try to avoid private invokes whenever possible because it feels we're doing a code smell.